src/main/java/org/apache/mesos/scheduler/DefaultScheduler.java

```diff
@@ -124,6 +129,31 @@ private void initializeRecoveryScheduler() {
                 recoveryStatusRef);
     }
 
+    private void initializeApi() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                JettyApiServer apiServer = new JettyApiServer(
```

Comment:
DefaultScheduler should schedule, and not bootstrap API interfaces. These are two separate concerns, and shouldn't be coupled to each other.

I would rather provide a DefaultSchedulerResourceRegistry or something similar, which provides a `List` of resources that default scheduler provides, and let the framework/service author use those in-addition to service/framework specific resources while bootstrapping Jetty and Jersey.